### PR TITLE
Add guide to use own S3 bucket with platform

### PR DESCRIPTION
**Pull request by IgorSusmelj**
Added the guide on how to use the Lightly platform with the S3 bucket
- This is a copy of the guide @japrescott wrote
- The resources will be fetched through the makefile as done with other parts of the docs with larger files

[The Lightly Platform — lightly 1.0.0 documentation.pdf](https://github.com/lightly-ai/lightly/files/7629398/The.Lightly.Platform.lightly.1.0.0.documentation.pdf)

### CODE REVIEW

File: docs/source/getting_started/platform.rst [hunk 1]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
 .. warning:: Keep the token for yourself and don't share it. Anyone with the
           token could access your datasets!
 
+
+How to use S3 with lightly
```
Comment (philippmwirth) [inline docs/source/getting_started/platform.rst:306]: Capital letter for web-app, right? "Lightly"

File: docs/source/getting_started/platform.rst [hunk 2]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
...
+------------------------------
+
+
+Lightly allows you to configure a remote datasource like Amazon S3 (Amazon Simple Storage Service) so that you don't need to upload your data to Lightly and can preserve its privacy.
+
+
+**What you will learn**
+
+
+In this guide we will show you how to setup your S3 bucket, configure your dataset to use said bucket and to only upload metadata to lightly while preserving the privacy of your data
```
Comment (philippmwirth) [inline docs/source/getting_started/platform.rst:316]: Lightly > lightly

File: docs/source/getting_started/platform.rst [hunk 3]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
...
+
+
+Setting up Amazon S3
+^^^^^^^^^^^^^^^^^^^^^^
+For Lightly to be able to create so-called `presigned URLs/read URLs <https://docs.aws.amazon.com/AmazonS3/latest/userguide/ShareObjectPreSignedURL.html>`_ to be used for displaying your data in your browser, Lightly needs at minimum to be able to read and list permissions on your bucket. If you want Lightly to create optimal thumbnails for you while uploading the metadata of your images, write permissions are also needed.
```
Comment (philippmwirth) [inline docs/source/getting_started/platform.rst:321]: optimal thumbnails? what do you mean?

Comment (philippmwirth) [reply docs/source/getting_started/platform.rst:321]: optional?

File: docs/source/getting_started/platform.rst [hunk 4]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
...
+
+Let us assume your bucket is called datalake. And let us assume the folder you want to use with Lightly is located at projects/farm-animals/
```
Comment (philippmwirth) [inline docs/source/getting_started/platform.rst:323]: Can you emphasize the *datalake*?

Comment (philippmwirth) [reply docs/source/getting_started/platform.rst:323]: Also make it more clear that the `projects/farm-animals` is the example dataset

Comment (philippmwirth) [reply docs/source/getting_started/platform.rst:323]: To me it was unclear why this is called farm animals

File: docs/source/getting_started/platform.rst [hunk 5]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
...
+1. Go to the `Identity and Access Management IAM page <https://console.aws.amazon.com/iamv2/home?#/users>`_ and create a new user for Lightly.
+2. Choose a unique name of your choice and select "Programmatic access" as "Access type". Click next
+    
+    .. figure:: resources/AWSCreateUser2.png
+        :align: center
+        :alt: Create AWS User
+
+        Create AWS User
+
+3. We will want to create very restrictive permissions for this new user so that it cant access other resources of your company. Click on "Attach existing policies directly" and then on "Create policy". This will bring you to new page
```
Comment (philippmwirth) [inline docs/source/getting_started/platform.rst:336]: can't > cant

Comment (philippmwirth) [reply docs/source/getting_started/platform.rst:336]: This will bring you to **a** new page

File: docs/source/getting_started/platform.rst [hunk 6]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
...
+
+        Permission policy in AWS
+5. Go to the next page and create tags as you see fit (e.g `external` or `lightly`) and give a name to your new policy prior to creating it.
+
+    .. figure:: resources/AWSCreateUser5.png
+        :align: center
+        :alt: Review and name permission policy in AWS
+
+        Review and name permission policy in AWS
+6. Return to the prior page and reload. Now when filtering policies your newly created policy will show up. Select it and continue setting up your new user.
```
Comment (philippmwirth) [inline docs/source/getting_started/platform.rst:382]: Which is the prior page here? Maybe mention it explicitly

File: docs/source/getting_started/platform.rst [hunk 7]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
...
+    
+    .. figure:: resources/AWSCreateUser6.png
+        :align: center
+        :alt: Attach permission policy to user in AWS
+
+        Attach permission policy to user in AWS
+7. Write down the `Access key ID` and the `Secret access key` in a secure location (such as a password manager) as you will not be able to access these information again (you can generate new keys and revoke old keys under `Security credentials` of a users detail page)
```
Comment (philippmwirth) [inline docs/source/getting_started/platform.rst:389]: these information**s** or this information

File: docs/source/getting_started/platform.rst [hunk 8]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
...
+    .. figure:: resources/AWSCreateUser7.png
+        :align: center
+        :alt: Get security credentials (access key id, secret access key) from AWS
+
+        Get security credentials (access key id, secret access key) from AWS
+
+**Preparing your data**
+
+
+For Lightly to be able to create embeddings and extract metadata from your data `lightly-magic` needs to be able to access your data. You can either download/sync your data from S3 or you can mount S3 as a drive. We recommend downloading your data from S3 as it makes the overall process faster.
```
Comment (philippmwirth) [inline docs/source/getting_started/platform.rst:400]: comma after "from your data"

File: docs/source/getting_started/platform.rst [hunk 9]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
...
+
+Uploading your data
+^^^^^^^^^^^^^^^^^^^^^^
+
+Create and configure a dataset
+
+1. `Create a new dataset <https://app.lightly.ai/dataset/create>`_ in Lightly
+2. Edit your dataset and select S3 as your datasource
+
+    .. figure:: resources/LightlyEdit1.png
```
Comment (philippmwirth) [inline docs/source/getting_started/platform.rst:420]: Nice, the picture here is important

File: docs/source/getting_started/platform.rst [hunk 10]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
...
+        :alt: Lightly S3 connection config
+        :width: 60%
+
+        Lightly S3 connection config
+
+6. Press save and ensure that at least the lights for List and Read turn green.
+
+
+Use Lightly
+Use `lightly-magic` and `lightly-upload` just as you always would with the following considerations;
```
Comment (philippmwirth) [inline docs/source/getting_started/platform.rst:445]: : > ;

**Review by philippmwirth — COMMENTED** [review]
I wonder if "datalake" is a good name for the dataset? I think it's rather confusing, why not name the dataset similar to the input directory, e.g. "farm-animals-dataset"? 

Left some comments, overall good work.

File: docs/source/getting_started/platform.rst (hunk 3 above)
Comment (japrescott) [reply docs/source/getting_started/platform.rst:321]: optimal in a sense that we know best how to make good thumbnails (size, compression, etc)

**Comment by IgorSusmelj** [comment]
Implemented the feedback and fixed some typos :)

File: docs/source/getting_started/platform.rst [hunk 11]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
...
+In this guide, we will show you how to setup your S3 bucket, configure your dataset to use said bucket, and only upload metadata to Lightly while preserving the privacy of your data
+
+
+Setting up Amazon S3
+^^^^^^^^^^^^^^^^^^^^^^
+For Lightly to be able to create so-called `presigned URLs/read URLs <https://docs.aws.amazon.com/AmazonS3/latest/userguide/ShareObjectPreSignedURL.html>`_ to be used for displaying your data in your browser, Lightly needs at minimum to be able to read and list permissions on your bucket. If you want Lightly to create optimal thumbnails for you while uploading the metadata of your images, write permissions are also needed.
+
+Let us assume your bucket is called `datalake`. And let us assume the folder you want to use with Lightly is located at projects/farm-animals/
+
+**Creating an IAM**
```
Comment (gewesp) [inline docs/source/getting_started/platform.rst:325]: Wouldn't that rather be 'setting up IAM'?

File: docs/source/getting_started/platform.rst [hunk 12]
```diff
@@ -302,3 +302,150 @@ account (top right)-> preferences on the
...
+
+**Downloading from S3 (recommended)**
+
+1. Install AWS cli by following the `guide of Amazon <https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html>`_
+2. Run `aws configure` and set the credentials
+3. Download/synchronize the folder located on S3 to your current directory `aws s3 sync s3://datalake/projects/farm-animals ./farm`
+
+**Mount S3 as a drive**
+
+Please follow official guidance or for Linux and MacOS use `s3fs-fuse <https://github.com/s3fs-fuse/s3fs-fuse>`_
```
Comment (gewesp) [inline docs/source/getting_started/platform.rst:410]: Would it be possible to link to the official guidance?  I'd also split it into two sentences. Period at end is missing.

Comment (IgorSusmelj) [reply docs/source/getting_started/platform.rst]: I updated the text and also added a reference to our guide for the s3fs-fuse setup :)
Thanks for the feedback!

File: docs/source/getting_started/platform.rst (hunk 3 above)
Comment (philippmwirth) [reply docs/source/getting_started/platform.rst:321]: I still don't think optimal makes any sense here 😄 but that's a minor thingy so you decide...

**Review by philippmwirth — APPROVED** [review]
This looks good now for a first version 👍

**Pull request merged**